packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart

```diff
@@ -170,9 +170,43 @@ class CpuProfileData with Serializable {
 
   factory CpuProfileData.fromJson(Map<String, Object?> json_) {
     final json = _CpuProfileDataJson(json_);
+
+    // All CPU samples.
+    final samples = json.traceEvents ?? [];
+
+    // Sort the samples so we can compute the observed time difference between
+    // each sample.
+    samples.sort((a, b) => a.timestampMicros!.compareTo(b.timestampMicros!));
```

Comment:
do the trace events already come in timestamp order?

Reply:
Unfortunately, no they don't seem to be (I tried putting an assert that all the diffs between samples were positive and it tripped on one of my larger profiles).